devito/compiler.py

```diff
@@ -194,12 +194,13 @@ def get_tmp_dir():
 
     :return: Path to a devito-specific tmp directory
     """
-    tmpdir = path.join(gettempdir(), "devito-%s" % getuid())
+    global _devito_compiler_tmpdir
```

Comment:
why does it need to be ``global``?

Reply:
Ping @ggorman

Reply:
Ping again, @ggorman. This is ready to land otherwise, so please respond.

Reply:
What I want here is for _devito_compiler_tmpdir to be globally cached so that it only creates one unique tmp dir per process. get_tmp_dir() is not a class member so I cannot instead use memoize.